File: LanguageTranslator/tests/Python/type_errors.py
# Test type checking.
def pos(n):
    return n
def fac():
	return "s"

i = 3
b = True
i = b
i = b*2 #error, boolean variable cannot be operand of arithmatic expression
i = False+1 #error, boolean value cannot be operand of arithmatic expression
i = "Hello"/2 #error, string value cannot be operand of arithmatic expression
i = 2,3 #error,disparity of values and variables
if b:
    i = pos(6)+1
if 1:
    i = i+fac() #error, string variable cannot be operand of arithmatic expression
    i = fac(b)  #error, fac() do not have parameters           
if i: #string variable cannot be the condition of if-else statement
    print(i)
while i: #string variable cannot be the condition of while statement
    print(i)
b = pos("1")
b = pos() #error, there should be parameters
b = pos(3,5,6) #error, there should be only one parameter
while pos(7):
    i = 1 

File: LanguageTranslator/tests/C/type_errors.c
// Test type checking.
int n = 'a'; //error, value 'a' is not int
_Bool c = 5; //error, value is not _Bool
_Bool pos(int n){
	return n;//error, return type conflicts with declaration
}
int fac(int n){
	return n;
}
char boo(char n){
	return n;
}
void main(){
	int i = 3;
	_Bool b = 1;
	i = i+1;
	i = b; //error, variable i has declared to be int
	i = b*2; //error, boolean variable cannot be operand of arithmatic expression
	b = i>0;
	if(b) printf(i);
	if(1) printf(i+1);
	b = pos('1'); //error, actual parameter conflict with formal parameter
	while(pos(7)){
		i = i+fac(5);
		i = i+boo('a'); //error, char variable cannot be operand of arithmatic expression
	}
}

